aiohomekit/controller/coap/pairing.py

```diff
@@ -0,0 +1,143 @@
+#
+# Copyright 2022 aiohomekit team
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import asyncio
+import logging
+
+from aiohomekit.controller.pairing import AbstractPairing
+from aiohomekit.model.characteristics.characteristic_types import CharacteristicsTypes
+from aiohomekit.model.services.service_types import ServicesTypes
+
+from .connection import CoAPHomeKitConnection
+
+logger = logging.getLogger(__name__)
+
+
+class CoAPPairing(AbstractPairing):
+    def __init__(self, controller, pairing_data):
+        super().__init__(controller)
+
+        self.connection = CoAPHomeKitConnection(
+            self, pairing_data["AccessoryIP"], pairing_data["AccessoryPort"]
+        )
+        self.connection_lock = asyncio.Lock()
+        self.pairing_data = pairing_data
+
+    async def _ensure_connected(self):
+        async with self.connection_lock:
+            if self.connection.is_connected:
+                return
+
+            await self.connection.connect(self.pairing_data)
+
+            if len(self.subscriptions):
+                logger.debug(
+                    "(Re-)subscribing to %d characteristics: %r"
+                    % (len(self.subscriptions), self.subscriptions)
+                )
+                await self.connection.subscribe_to(list(self.subscriptions))
+
+        return
+
+    async def close(self):
+        return
+
+    def event_received(self, event):
+        self._callback_listeners(event)
+
+    def _callback_listeners(self, event):
+        for listener in self.listeners:
+            try:
+                logger.debug(f"callback ev:{event!r}")
+                listener(event)
+            except Exception:
+                logger.exception("Unhandled error when processing event")
+
+    async def identify(self):
+        await self._ensure_connected()
+        return await self.connection.do_identify()
+
+    async def list_accessories_and_characteristics(self):
+        await self._ensure_connected()
+
+        accessories = await self.connection.get_accessory_info()
+
+        for accessory in accessories:
+            for service in accessory["services"]:
+                try:
+                    service["type"] = ServicesTypes.get_uuid(service["type"])
```

Comment:
If you rebase this is gone in favour of `aiohomekit.uuid.normalize_uuid`. It won't raise on unknown UUID's so you can lose the KeyError too.